tests/test_estimate_params.py

```diff
@@ -0,0 +1,965 @@
+        # This test is also valdiating if we can see which run has a wrong error.
+        with self.assertRaises(ValueError) as cm:
+            m.estimate_params(times=incorrectTimesLen, inputs=inputs, outputs=outputs)
+        self.assertEqual(
+            'Times, inputs, and outputs must be same length for the run at index 1. Length of times: 5, Length of inputs: 4, Length of outputs: 4',
+            str(cm.exception)
+        )
+
+        # Incorrect lengths for times.
+        with self.assertRaises(ValueError):
+            m.estimate_params(times=[time1, [time2]], inputs=inputs, outputs=outputs)
+        self.assertEqual(
+            'Times, inputs, and outputs must be same length for the run at index 1. Length of times: 5, Length of inputs: 4, Length of outputs: 4',
+            str(cm.exception)
+        )
+        # Wrapper list becomes a set.
+        timesSet = [time1, time2]
+        # Unhashable type for outputs
+        with self.assertRaises(TypeError):
+            m.estimate_params(times=set(timesSet), inputs=inputs, outputs=set(outputs))
+        
+        time1 = np.array([0, 1, 2, 4, 5, 6, 7, 8, 9])
+        time2 = [0, 1, 2, 3]
+
+        # Confirming estimate_params works when different runs are passed in as different data types.
+        # Passing in time1 as a np.array datatype.
+        m.estimate_params(times=[time1, time2], inputs=inputs, outputs=outputs)
+
+        time1 = [0, 1, 2, 4, 5, 6, 7, 8, 9]
+        time2 = [0, 1, 2, 3]
+
+        inputs = [[{}]*9, [{}]*4]
+        outputs = [[{'x': 1.83},
+            {'x': 36.95},
+            {'x': 62.36},
+            {'x': 77.81},
+            {'x': 83.45},
+            {'x': 79.28},
+            {'x': 65.3},
+            {'x': 41.51},
+            {'x': 7.91},], 
+            np.array([
+                {'x': 1.83},
+                {'x': 36.95},
+                {'x': 62.36},
+                {'x': 77.81},
+            ])]
+
+        # Passing in an np.array structure as a runs for outputs.
+        m.estimate_params(times=[time1, time2], inputs=inputs, outputs=outputs)
+
+        # Test case that would be fixed with future changes to Containers
+        # with self.assertRaises(ValueError):
+        #     m.estimate_params(times=[incorrectTimesLen], inputs=[inputs], outputs=[outputs])
+
+
+    def test_incorrect_lens(self):
+        """
+        Goal of this test is to compare the results of passing in incorrect lenghts into our estimate_params call.
+
+        Furthermore, checks incorrect lenghts within multiple runs.
+        """
+        # Initalizing our model
+        m = ThrownObject(process_noise = 0, measurement_noise = 0)
+        results = m.simulate_to_threshold(save_freq=0.5)
+        gt = m.parameters.copy()
+
+        # Defined Keys and Bounds for estimate_params
+        keys = ['thrower_height', 'throwing_speed']
+        
+        m = ThrownObject()
```

Comment:
Do we want this repeat call here?

Reply:
Nope, we do not want these here. Removed it.